Mathlib/Algebra/Lie/Killing.lean

```diff
@@ -537,63 +557,176 @@ lemma killingForm_apply_eq_zero_of_mem_rootSpace_of_add_ne_zero {α β : H → K
     (LieSubmodule.iSup_eq_top_iff_coe_toSubmodule.mp <| iSup_weightSpace_eq_top K H L)
   exact LinearMap.trace_eq_zero_of_mapsTo_ne hds σ hσ hf
 
+/-- Elements of the `α` root space which are Killing-orthogonal to the `-α` root space are
+Killing-orthogonal to all of `L`. -/
+lemma mem_ker_killingForm_of_mem_rootSpace_of_forall_rootSpace_neg
+    {α : H → K} {x : L} (hx : x ∈ rootSpace H α)
+    (hx' : ∀ y ∈ rootSpace H (-α), killingForm K L x y = 0) :
+    x ∈ LinearMap.ker (killingForm K L) := by
+  rw [LinearMap.mem_ker]
+  ext y
+  have hy : y ∈ ⨆ β, rootSpace H β := by simp [iSup_weightSpace_eq_top K H L]
+  induction hy using LieSubmodule.iSup_induction'
+  · next β y hy =>
+    by_cases hαβ : α + β = 0
+    · exact hx' _ (add_eq_zero_iff_neg_eq.mp hαβ ▸ hy)
+    · exact killingForm_apply_eq_zero_of_mem_rootSpace_of_add_ne_zero K L H hx hy hαβ
+  · simp
+  · simp_all
+
 namespace IsKilling
 
 variable [IsKilling K L]
 
+variable {K L} in
+/-- The restriction of the Killing form to a Cartan subalgebra, as a linear equivalence to the
+dual. -/
+@[simps!]
+noncomputable def cartanEquivDual :
+    H ≃ₗ[K] Module.Dual K H :=
+  (traceForm K H L).toDual <| traceForm_cartan_nondegenerate K L H
+
+/-- This is Proposition 4.18 from [carter2005] except that we use
+`LieModule.exists_forall_lie_eq_smul` instead of Lie's theorem (and so avoid
+assuming `K` has characteristic zero). -/
+lemma cartanEquivDual_symm_apply_mem_corootSpace (α : Weight K H L) :
+    (cartanEquivDual H).symm α ∈ corootSpace α := by
+  obtain ⟨e, he₀, he⟩ := exists_forall_lie_eq_smul K H L α
```

Comment:
Would you mind writing out the type of the existential, to improve readability a bit. (It's not that important, but since this is the step where you avoid char 0, I think it doesn't hurt to emphasize it.)

Reply:
Good idea, done!